x/metadata/keeper/keeper_test.go

```diff
@@ -312,12 +316,203 @@ func (s *KeeperTestSuite) TestValidateAllOwnerPartiesAreSigners() {
+			granter:               nil,
+			grantee:               nil,
+			errorMsg:              fmt.Sprintf("missing signature from [%s (PARTY_TYPE_OWNER)]", s.user2),
+		},
+	}
+
+	now := s.ctx.BlockHeader().Time
+	s.Require().NotNil(now)
+
+	// Test cases
+	for _, tc := range cases {
+		s.T().Run(tc.name, func(t *testing.T) {
+			createAuth := tc.grantee != nil && tc.granter != nil
+			if createAuth {
+				a := authz.NewCountAuthorization(tc.msgTypeURL, tc.allowedAuthorizations)
+				err := s.app.AuthzKeeper.SaveGrant(s.ctx, tc.grantee, tc.granter, a, now.Add(time.Hour))
+				s.Require().NoError(err)
+			}
+
+			err := s.app.MetadataKeeper.ValidateAllPartiesAreSignersWithAuthz(s.ctx, tc.owners, tc.signers, tc.msgTypeURL)
+			if len(tc.errorMsg) == 0 {
+				assert.NoError(t, err, "%s unexpected error", tc.name)
+			} else {
+				assert.EqualError(t, err, tc.errorMsg, "%s error", tc.name)
+			}
+
+			// validate allowedAuthorizations
+			if err == nil {
+				auth, _ := s.app.AuthzKeeper.GetCleanAuthorization(s.ctx, tc.grantee, tc.granter, tc.msgTypeURL)
+				if tc.allowedAuthorizations == 1 {
+					// authorization is deleted after one use
+					assert.Nil(t, auth)
+				} else {
+					actual := auth.(*authz.CountAuthorization).AllowedAuthorizations
+					assert.Equal(t, tc.allowedAuthorizations-1, actual)
+				}
+			}
+		})
+	}
+
+	// test cases with owners that have different signers
+
+	type grant struct {
+		allowedAuthorizations int32
+		granter               sdk.AccAddress
+		grantee               sdk.AccAddress
+		errorMsg              string
+	}
+
+	tests2 := []struct {
+		name       string
+		owners     []types.Party
+		signers    []string
+		msgTypeURL string
+		grants     []grant
+	}{
+		// test with two owners (1 & 2), and two different signers (3 & 4),
+		// with two authz count authorization
+		//	- grants:
+		//		granter: 1, grantee: 3, count: 1
+		//		granter: 2, grantee: 4, count: 2
+		//
+		//	Then a message is executed, and we check that both grants have updated appropriately.
+		//	Then another message is executed, and we make sure it fails (because the first grant only had 1 use),
+		//	and that the second grant still exists with 1 use left.
+		{
+			name: "two parties - two different signers - with authz count grants",
+			owners: []types.Party{
+				{Address: s.user1, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user2, Role: types.PartyType_PARTY_TYPE_OWNER}},
+			signers:    []string{s.user3, s.user4},
+			msgTypeURL: types.TypeURLMsgDeleteScopeRequest,
+			grants: []grant{
+				{
+					allowedAuthorizations: 1,
+					granter:               s.user1Addr,
+					grantee:               s.user3Addr,
+					errorMsg:              "",
+				},
+				{
+					allowedAuthorizations: 2,
+					granter:               s.user2Addr,
+					grantee:               s.user4Addr,
+					errorMsg:              fmt.Sprintf("missing signatures from [%s (PARTY_TYPE_OWNER) ]", s.user1),
+				},
+			},
+		},
+	}
```

Comment:
> Also, I'm not sure if this is covered by any of the other tests, but it'd probably be good to have a test where two signers are required (1 & 2), and there are two grants (1 -> 3 for one use and 2 -> 3 for two uses), and only 3 is the signer; make sure both grants are updated/deleted appropriately.

This is very similar to the ^^^ case that we now want to move to `x/metadata/client/cli/cli_test.go`. Instead of:

```
```